### ISSUE TITLE
hidden quests

### ISSUE THREAD

**Issue by Jibec** [westnordost/StreetComplete#1304]
Many quests are hidden until you commit your changes.

Example for a building
You need first to tell the building usage,
Commit
Then you can tell the number of levels
Commit
Then you can tell the roof type

The issue is I don't allow instant upload, because I don't want to tell on real time where I am, or don't want to give to many indication on where I go. So I mostly commit once per day.

The impact of this is forcing me to visit a place multiple time, and there is many place I'll probably visit only once.

**How to Reproduce**
Find a zone with a building but no additional tags.

**Versions affected**
9.0

Thanks for this great tool and your work

**Comment by Jibec** [comment]
Indeed, feel free to close this issue.

To contribute the discussion, I think a single symbol counting the number of quests is sufficient, and it should open a menu with the list of question on the same object.

**Comment by ENT8R** [comment]
I can't close the issue, you have to do this :smile:

**Comment by westnordost** [comment]
It is not a duplicate of #124, however, I would still have closed this because to implement this feature would be too complex for the positive effect.

In the cases that @Jibec mentioned, the quest types depend on each other:
The quest for roof shape is only shown after the number of building (roof) levels has been answered. The quest for building levels is in turn only asked after the building type quest has been answered.
In other words, the roof shape quest is only shown for buildings that have `roof:levels` set, which is set by the building levels quest.

So, the roof shape quest is not shown because the quests are always based on the version of the element as downloaded from OSM. They are **not** based on the to-be-uploaded elements stored on the phone because the answers made by the user in StreetComplete are actually not stored as OSM elements but as diffs. Changing this would make conflict-resolution somewhat complex.
